### PR TITLE
Added multi download support to pipeline artifacts

### PR DESCRIPTION
**Pull request by owenhuynMSFT**

### CODE REVIEW

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs [hunk 1]
```diff
@@ -48,85 +49,96 @@ public class PipelineArtifactServer
...
+            if (string.IsNullOrEmpty(downloadParameters.ArtifactName))
             {
-                throw new ArgumentException($"The artifact is not of the type Pipeline Artifact. Unrecognized type: {art.Resource.Type}.");
+                List<BuildArtifact> artifacts = await buildHelper.GetArtifactsAsync(downloadParameters.ProjectId, downloadParameters.BuildId, cancellationToken);
+                IEnumerable<BuildArtifact> pipelineArtifacts = artifacts.Where(a => a.Resource.Type == PipelineArtifactTypeName);
+                if (pipelineArtifacts.Count() == 0)
+                {
+                    throw new ArgumentException("Could not find any pipeline artifacts in the build.");
+                }
+                else
```
Comment (jahsu-MSFT) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs]: The else statement shouldn't be necessary here

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs [hunk 2]
```diff
@@ -48,85 +49,96 @@ public class PipelineArtifactServer
...
+                {
+                    buildArtifact = await buildHelper.GetArtifact(downloadParameters.ProjectId, downloadParameters.BuildId, downloadParameters.ArtifactName, cancellationToken);
+                }
+                else if (downloadParameters.ProjectRetrievalOptions == BuildArtifactRetrievalOptions.RetrieveByProjectName)
+                {
+                    if (string.IsNullOrEmpty(downloadParameters.ProjectName))
+                    {
+                        throw new InvalidOperationException("Project name can't be empty when trying to fetch build artifacts!");
+                    }
+                    else
```
Comment (jahsu-MSFT) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs]: Same here. The else statement is redundant.

Comment (owenhuynMSFT) [reply src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs]: I agree explicit else better. Plus it forces the developer to think of all cases.

Comment (jahsu-MSFT) [reply src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs]: The coding style is known as guard block and is common to be used to return/throw exception early if something went wrong (wrong parameter or invalid state). Adding a else statement will also work but it's redundant IMO.

**Pull request merged**